src/Umbraco.Core/IO/FileSystems.cs

```diff
@@ -282,7 +282,7 @@ private ShadowWrapper CreateShadowWrapper(IFileSystem filesystem, string shadowP
         {
             lock (_shadowLocker)
             {
-                var wrapper = new ShadowWrapper(filesystem, _ioHelper, _hostingEnvironment, _loggerFactory, shadowPath, IsScoped);
+                var wrapper = new ShadowWrapper(filesystem, _ioHelper, _hostingEnvironment, _loggerFactory, shadowPath,() => IsScoped());
```

Comment:
I have no idea how this worked previously.

Reply:
>  System.UnauthorizedAccessException : File 'f1.txt' is outside this filesystem's root.

Doesn't pass on dev ops pipeline, fantastic...

Reply:
🤦 - I will try to help looking into why..

Reply:
I have set up an additional pipeline, that just filters the tests with `--filter "FullyQualifiedName~ScopeFileSystemsTests"`

https://dev.azure.com/umbraco/Umbraco%20Cms/_build/results?buildId=50804&view=logs&j=fd490c07-0b22-5182-fac9-6d67fe1e939b

Reply:
what a star, on a separate note, I wonder how much faster the pipeline would be if system.debug was set to false

Reply:
I changed the `system.debug` to `false`.. No matter how much time better time it gives, it don't make sense to have it on as default.

Reply:
> File original: [f1.txt] full: [C:\Users\VssAdministrator\AppData\Local\Temp\UmbracoTemp\media\f1.txt] is outside this filesystem's root.

Reply:
That seems completely reasonable, it's combines correctly, I can't see why _ioHelper.PathStartsWth would be falsey

Reply:
`_rootPath`  seems to be 'D:\a\1\s\src\Umbraco.Tests.Integration\App_Data\TEMP\ShadowFs\0fxfhwos\media'

Reply:
Added this naive logging: https://github.com/umbraco/Umbraco-CMS/commit/7a40fc6640de4da061b3bf0a630efab129fa412b

Output

```
 info: Umbraco.Core.IO.PhysicalFileSystem[0]
       GetFullPath 1: opath='f1.txt',path='f1.txt', _rootPath='D:\a\1\s\src\Umbraco.Tests.Integration\App_Data\TEMP\ShadowFs\0fxfhwos\media'
 info: Umbraco.Core.IO.PhysicalFileSystem[0]
       GetFullPath 2: opath='f1.txt',path='f1.txt', _rootPath='D:\a\1\s\src\Umbraco.Tests.Integration\App_Data\TEMP\ShadowFs\0fxfhwos\media'
 info: Umbraco.Core.IO.PhysicalFileSystem[0]
       GetFullPath 3: opath='f1.txt',path='f1.txt', _rootPath='D:\a\1\s\src\Umbraco.Tests.Integration\App_Data\TEMP\ShadowFs\0fxfhwos\media'
 info: Umbraco.Core.IO.PhysicalFileSystem[0]
       GetFullPath 4: opath='f1.txt',path='D:\a\1\s\src\Umbraco.Tests.Integration\App_Data\TEMP\ShadowFs\0fxfhwos\media\f1.txt', _rootPath='D:\a\1\s\src\Umbraco.Tests.Integration\App_Data\TEMP\ShadowFs\0fxfhwos\media'
 info: Umbraco.Core.IO.PhysicalFileSystem[0]
       GetFullPath 5: opath='f1.txt',path='D:\a\1\s\src\Umbraco.Tests.Integration\App_Data\TEMP\ShadowFs\0fxfhwos\media\f1.txt', _rootPath='D:\a\1\s\src\Umbraco.Tests.Integration\App_Data\TEMP\ShadowFs\0fxfhwos\media'
 info: Umbraco.Core.IO.PhysicalFileSystem[0]
       GetFullPath 6: opath='f1.txt',path='D:\a\1\s\src\Umbraco.Tests.Integration\App_Data\TEMP\ShadowFs\0fxfhwos\media\f1.txt', _rootPath='D:\a\1\s\src\Umbraco.Tests.Integration\App_Data\TEMP\ShadowFs\0fxfhwos\media'
 info: Umbraco.Core.Migrations.Install.DatabaseSchemaCreator[0]
       Create Table umbracoUser (-1): 

```

Reply:
I gotta do some work so I have to bail, but could you merge edd562b into netcore/bugfix/scopeFileSystemTests_fix_on_azure_pipelines

Wondering if that will help shine light on issue.

Reply:
>       GetFullPath 7: opath='f1.txt',path='C:\Users\VssAdministrator\AppData\Local\Temp\UmbracoTemp\media\f1.txt', _rootPath='C:\Users\VSSADM~1\AppData\Local\Temp\UmbracoTemp\media'


I guess the issue is `VSSADM~1 != VssAdministrator`

Reply:
Indeed, so we could clean up root path before comparing or resolve the difference in GlobalSettings

Reply:
I think we should call `Path.GetFullPath` when we set `_rootPath` in the ctor..

Reply:
thanks so much for your help on this, you prevented so many tears, test passed in  14cc5ae assume it will in 70b7ef8 also

Reply:
yep, hype

Reply:
I'm the one who thanks for your massive help - and patience 💪